diff --git a/src/ws-egl.cpp b/src/ws-egl.cpp
index b91e4dd..6f3f604 100644
--- a/src/ws-egl.cpp
+++ b/src/ws-egl.cpp
@@ -29,6 +29,7 @@
 #include <epoxy/egl.h>
 #include <cassert>
 #include <cstring>
+#include <cstdio>

 #ifndef EGL_WL_bind_wayland_display
 #define EGL_WAYLAND_BUFFER_WL 0x31D5
@@ -304,8 +305,13 @@ void ImplEGL::foreachDmaBufModifier(std::function<void (int format, uint64_t mod
             modifiers[0] = DRM_FORMAT_MOD_INVALID;
         }

-        for (int j = 0; j < numModifiers; j++)
+        for (int j = 0; j < numModifiers; j++) {
+            if (modifiers[j] != DRM_FORMAT_MOD_LINEAR) {
+                continue;
+            }
+
             callback(formats[i], modifiers[j]);
+        }
     }
 }

